lib/src/dynamics/joints/joint.dart

```diff
@@ -3,32 +3,35 @@ import '../../../forge2d.dart';
 /// The base joint class. Joints are used to constrain two bodies together in various fashions. Some
 /// joints also feature limits and motors.
 abstract class Joint {
-  static Joint create(World world, JointDef def) {
-    switch (def.type) {
-      case JointType.mouse:
-        return MouseJoint(def as MouseJointDef);
-      case JointType.distance:
-        return DistanceJoint(def as DistanceJointDef);
-      case JointType.prismatic:
-        return PrismaticJoint(def as PrismaticJointDef);
-      case JointType.revolute:
-        return RevoluteJoint(def as RevoluteJointDef);
-      case JointType.weld:
-        return WeldJoint(def as WeldJointDef);
-      case JointType.friction:
-        return FrictionJoint(def as FrictionJointDef);
-      case JointType.wheel:
-        return WheelJoint(def as WheelJointDef);
-      case JointType.gear:
-        return GearJoint(def as GearJointDef);
-      case JointType.pulley:
-        return PulleyJoint(def as PulleyJointDef);
-      case JointType.constantVolume:
-        return ConstantVolumeJoint(world, def as ConstantVolumeJointDef);
-      case JointType.rope:
-        return RopeJoint(def as RopeJointDef);
-      case JointType.motor:
-        return MotorJoint(def as MotorJointDef);
+  static T create<T extends Joint>(World world, JointDef def) {
+    if (T == ConstantVolumeJoint || def.type == JointType.constantVolume) {
```

Comment:
The `type` should probably be removed from `JointDef`, and the actual type can just be compared here, and have an and condition in the statements, instead of an or.
But that can be done in a follow-up, if you don't want to make this PR larger.

Reply:
Indeed wanted to do it as a follow up to resolve #43